### ISSUE TITLE
Doc formatting is broken in various places/ways

### ISSUE THREAD

**Issue by cluhmann** [pymc-labs/pymc-marketing#367]
The list of methods for the MMM module is squashed horizontally. Many methods have long names, but I am not sure this is the problem.  I am viewing on a desktop with a widescreen, so I don't think it's a resolution/aspect ratio issue either.  Here is a screenshot of the `delayed_saturated_mmm` module found [here](https://www.pymc-marketing.io/en/stable/api/generated/pymc_marketing.mmm.delayed_saturated_mmm.DelayedSaturatedMMM.html#pymc_marketing.mmm.delayed_saturated_mmm.DelayedSaturatedMMM):

![image](https://github.com/pymc-labs/pymc-marketing/assets/39170995/006d58d2-9ba2-4ced-8fd5-3ee4c706eda8)

In addition, some of the MMM methods have docstrings that I suspect are incorrectly formatted.  Here is an example of the documentation of the `channel_contributions_forward_pass()` method.  The return values are formatted correctly, but the arguments (`channel_data`) are not:

![image](https://github.com/pymc-labs/pymc-marketing/assets/39170995/e1311d01-77bb-4519-a8b9-6a629fc8daf8)

The same problems occur with [`get_channel_contributions_forward_pass_grid()`](https://www.pymc-marketing.io/en/stable/api/generated/pymc_marketing.mmm.delayed_saturated_mmm.DelayedSaturatedMMM.get_channel_contributions_forward_pass_grid.html#pymc_marketing.mmm.delayed_saturated_mmm.DelayedSaturatedMMM.get_channel_contributions_forward_pass_grid).

**Comment by wd60622** [comment]
Yes, I've noticed this too in a spots. Usually fixed by adding a space the docstring in order to separate the sections better. 
There might be a way to check and/or format automatically

**Comment by wd60622** [comment]
Running list: 

## [BaseMMM](https://www.pymc-marketing.io/en/stable/api/generated/pymc_marketing.mmm.base.BaseMMM.html#pymc_marketing.mmm.base.BaseMMM)
- [ ] BaseMMM.compute_channel_contribution_original_scale
- [ ] BaseMMM.get_target_transformer
- [ ] BaseMMM.plot_channel_contribution_share_hdi
- [ ] BaseMMM.plot_channel_parameter
- [ ] BaseMMM.plot_component_contributions
- [ ] BaseMMM.plot_prior_predictive

## [MMM](https://www.pymc-marketing.io/en/stable/api/generated/pymc_marketing.mmm.base.MMM.html#pymc_marketing.mmm.base.MMM)

The ones from above plus some more

- [ ] MMM.compute_channel_contgribution_original_scale
- [ ] MMM.get_target_transformer
- [ ] MMM.plot_channel_contribution_share_hdi
- [ ] MMM.plot_channel_parameter
- [ ] MMM.plot_component_contributions
- [ ] MMM.plot_prior_predictive
- [ ] MMM.validate_channel_columns
- [ ] MMM.validate_date_col
- [ ] MMM.validate_target

**Comment by wd60622** [comment]
@juanitorduz 
You may have hit many of these when doing the ruff checks

**Comment by anevolbap** [comment]
This looks resolved.

Two things happened that together address what was reported here:

1. The classes named in the original report (`DelayedSaturatedMMM`, `BaseMMM`) were removed in #2487. The running list in the May 2024 comment refers to classes that no longer exist.

2. The underlying cause noted by @OriolAbril in January 2024 ("Sphinx prints a lot of warnings... about blank lines or indents missing mostly") was the broader problem tracked by #1210, and that one was just resolved in #2547. The sphinx warning log on `main` went from 964 distinct warnings to 0, and #2547 also added an enforcing docs CI gate (`sphinx-build -W` in GH Actions plus `sphinx.fail_on_warning: true` on Read the Docs). Any new docstring with missing blank lines or wrong indents will now fail the PR and cannot land.

Safe to close.
